### PR TITLE
refactor(utils): extract redux extension logic into wrapper functions

### PR DESCRIPTION
**Pull request by PrettyCoffee**
Chunk of this PR: https://github.com/jotaijs/jotai-devtools/pull/32

This will create a redux devtools extension wrapper to make it easier to use and prevent the repetition of the code.
Furthermore adds some documentation and links for better DX.

It should not change any functionality but only do some refactoring.

### CODE REVIEW

**Comment by dai-shi** [comment]
Thanks for working on this.

Just to confirm and be on the same page, I hoped #33 would fix it, but `extention.disconnect()` closing all connections prevents it, correct?
I wonder if this is an issue with Redux tools, and if we report it, there's possibility that they can fix (so that #33 works).

File: src/utils/redux-extension/index.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export * from './getReduxExtension';
+export * from './createReduxConnection';
```
Comment (dai-shi) [inline src/utils/redux-extension/index.ts:2]: This is just a preference of mine, but I'd avoid `index` stuff.
Can you simply import them directly?

**Review by dai-shi — COMMENTED** [review]
Other than that, the refactoring looks okay.

**Comment by PrettyCoffee** [comment]
> Thanks for working on this.

I am happy to help :)

> Just to confirm and be on the same page, I hoped #33 would fix it, but `extention.disconnect()` closing all connections prevents it, correct?

Yes, I personally think you shouldn't merge #33. When unrendering a single `useAtomDevtools` it will disconnect all of them and not the one where it is used.

>  I wonder if this is an issue with Redux tools, and if we report it, there's possibility that they can fix (so that #33 works).

Actually no, that's not bug, it is the expected behavior.

The extension itself owns all connections:

image

When calling `extension.disconnect()`, all connections should be disconnected.

The only thing we could do is creating a feature request for disconnecting single connections, like `connection.disconnect()` or `extension.disconnect(connection)` if that makes sense.
Not sure what their thoughts are on that since redux uses the extension completely differently.

I can create a ticket for that and reference it 👍

**Review by dai-shi — APPROVED** [review]
As long as this is just refactoring without changing logic, it should be safe to merge.

**Review by arjunvegda — APPROVED** [review]
Thanks a ton again for your work!

**Pull request merged**